unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,108 +24,170 @@ interface UpdateReferralFeeProps {
-    })
+  const getLock = async () => {
+    const service = new SubgraphService()
+    return service.lock(
+      {
+        where: {
+          id: lockAddress,
+        },
+      },
+      {
+        network,
+      }
+    )
   }
 
   const setReferrerFeeMutation = useMutation(setReferrerFee)
 
+  const isValidAddress = ethers.utils.isAddress(referralAddress)
+
   const onSubmit = async (fields: FormProps) => {
-    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync(fields)
+    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync({
+      ...fields,
+      referralAddress,
+    })
 
     await ToastHelper.promise(setReferrerFeePromise, {
-      loading: 'Updating referrer fee',
+      loading: 'Setting new referrer',
       error: 'Failed to update the values, please try again.',
-      success: 'Referrer fee updated.',
+      success: 'Referrer set',
     })
-  }
 
-  const { isLoading, data: referralFeePercentage } = useQuery(
-    ['getReferrerFees', lockAddress, network, setReferrerFeeMutation.isSuccess],
-    async () => getReferrerFees()
+    reset()
+  }
+  const { isLoading, data: referralFeesData } = useQuery(
+    ['getLock', lockAddress, network, setReferrerFeeMutation.isSuccess],
+    async () => getLock()
   )
 
-  useEffect(() => {
-    setValue('referralFeePercentage', (referralFeePercentage ?? 0) / 100)
-    setReferralFee((referralFeePercentage ?? 0) > 0)
-  }, [referralFeePercentage])
+  const referralFees = referralFeesData?.referrerFees || []
 
-  const disabledInput =
+  const isDisabledReferrerInput =
     disabled || setReferrerFeeMutation.isLoading || isLoading
 
   return (
     <form className="grid gap-6" onSubmit={handleSubmit(onSubmit)}>
       <div className="grid gap-2">
-        <ToggleSwitch
-          title="Referrer fee %"
-          enabled={referralFee}
-          disabled={disabledInput}
-          setEnabled={(enabled) => {
-            setReferralFee(enabled)
-            setValue(
-              'referralFeePercentage',
-              enabled ? (referralFeePercentage ?? 0) / 100 : 0
-            )
-          }}
-        />
         <Input
           type="number"
           {...register('referralFeePercentage', {
             valueAsNumber: true,
-            min: 0,
+            min: 1,
             max: 100,
           })}
           error={
             errors?.referralFeePercentage &&
-            'This field accept percentage value between 0 and 100.'
+            'This field accept percentage value between 1 and 100.'
           }
-          disabled={disabledInput || !referralFee}
+          disabled={isDisabledReferrerInput}
+        />
+      </div>
+
+      <div className="grid gap-2">
+        <ToggleSwitch
+          title="Referrer address"
+          enabled={isReferralAddressEnabled}
+          disabled={isDisabledReferrerInput}
+          setEnabled={(enabled: boolean) => {
+            setIsReferralAddressEnabled(enabled)
+          }}
+        />
+
```

Comment:
Right. This is supposed to be a boolean whenever we toggle the referrer address field. I renamed this to `isReferralAddressToggled` so it would make more sense ✅